### ISSUE TITLE
Improve stacktrace for failed dependency loading

### ISSUE THREAD

**Issue by guybedford** [ModuleLoader/es6-module-loader#397]
Originally posed by @OliverJAsh in https://github.com/systemjs/systemjs/issues/512:

This is what the errors looks like when a dependency fails to load:

```
Potentially unhandled rejection [3] Internal Server Error: <dependency>
	Error loading <dependency> from <requester>
	Error loading <dependency> (WARNING: non-Error used)
```

Sometimes `dependency` is a nested dependency of a dependency of `requester`, e.g. `requester -> foo -> bar -> dependency`. Is it possible to show the immediate requester (`bar`)?

**Comment by nyarly** [comment]
Looking at an issue in systemjs/builder, I found that the right way to address that would be here. I think load objects would need to be extended to include their (first?) immediate requester, and their un-normalized requested name (i.e. "requested as") in order to report these errors usefully.

I've started (i.e. 15 minutes to fork, clone, write a failing test) work on a PR for this.

**Comment by guybedford** [comment]
I think this can be done via some reflective lookups in the error handler itself to run a dependency trace just for the error message.

**Comment by guybedford** [comment]
Thanks @nyarly for your work on this.

**Comment by guybedford** [comment]
Released in 0.17.4.
